src/map.h

```diff
@@ -1703,7 +1703,8 @@ class map
         tripoint_range points_in_rectangle( const tripoint &from, const tripoint &to ) const;
         tripoint_range points_in_radius( const tripoint &center, size_t radius, size_t radiusz = 0 ) const;
 
-        std::list<item_location> get_active_items_in_radius( const tripoint &center, int radius ) const;
+        std::list<item_location> get_active_items_in_radius( const tripoint &center, int radius,
+                std::string type = "" ) const;
```

Comment:
```suggestion
                cata::optional<std::string> type = cata::nullopt ) const;
```

Avoids having an undocumented sentinel value that triggers special behavior.

Btw. why a string and not an enumeration? It seems only few (two in fact) values are appropriate for this.

Reply:
No good reason, will change it to an enum.